ykrt/src/compile/jitc_yk/codegen/x86_64/lsregalloc.rs

```diff
@@ -1013,6 +1035,9 @@ pub(crate) enum RegConstraint<R: Register> {
     Output,
     /// The result of this instruction will be stored in register `R`.
     OutputFromReg(R),
+    /// A temporary register *x*: it will be clobbered by the instruction and left in an
+    /// indeterminate state on exit.
```

Comment:
Does this mean a temporary is immediately "unallocated" before the next instruction?

Reply:
If it spills, where is the original value restored, or does it not work like that?

Reply:
Let me ask more precisely: If requesting a temporary register causes a spill, don't we need to restore the spilled value afterwards? If so, where does this happen?

Reply:
The register allocator handles spilling: it returns to the caller a register they can use, and in a sense that's all they have to worry about. `Temporary` is not dissimilar to `Output`: it spills (if necessary) anything already in that register; and then you can write to that register to your heart's content.

Reply:
Ah, so once something spills, it's spilled for good?

Reply:
Yes. It means we spill each SSA value at most once.

Reply:
Interesting. That's a design I hadn't considered.

Does this extend to calls? Do we evict potentially frequently-used variables to the stack for caller save?

Reply:
Yes, we clobber according to the SysV ABI.